src/main/java/org/jenkinsci/plugins/codescene/Domain/DeltaAnalysisResult.java

```diff
@@ -50,14 +52,26 @@ private static Warnings warningsFrom(JsonObject deltaResult) {
         return ws;
     }
 
+    private RiskDescription descriptionOfRiskFrom(JsonObject deltaResult, final String version) {
+        if (version.equals("1")) {
+            return new RiskDescription("No risk description available: upgrade CodeScene");
+        }
+
+        return new RiskDescription(deltaResult.getString("description"));
+    }
+
     private void ensureTheVersionIsSupported(JsonObject result) {
-        final String version = result.getString("version");
+        final String version = versionOf(result);
 
-        if (!version.equals("1")) {
-            throw new RuntimeException("The CodeScene API reports version " + version + ", which we don't know. You need to upgrade this plug-in.");
+        if (!version.equals("1") || !version.equals("2")) {
```

Comment:
This never matches. I guess it should have been sth. like this instead:
```suggestion
        if (! (version.equals("1") || version.equals("2"))) {
```
Moreover, it seems that the previous version "You need to upgrade this plugin" was better than the current "You need to upgrade CodeScene".

Reply:
One more note. Although it's probably obvious (and intended) this doesn't work when user has a new CodeScene returning delta api version 2 and old jenkins plugin version. In that case, he gets this error:
```
ERROR: Build step failed with exception
java.lang.RuntimeException: The CodeScene API reports version 2, which we don't know. You need to upgrade this plug-in.
```

Reply:
Yes, that's the intended behavior -- the new plugin should be backwards compatible, but in case CodeScene is upgraded the user is expected to upgrade the Jenkins plugin too.